src/Foreign/JavaScript/TH.hs

```diff
@@ -186,6 +186,9 @@ instance MonadAtomicRef m => MonadAtomicRef (WithWebView x m) where
 withWebViewSingleton :: WebView -> (forall x. WebViewSingleton x -> r) -> r
 withWebViewSingleton wv f = f $ WebViewSingleton wv
```

Comment:
What if we just gave it an explicit type in here, but still hid it from the user? e.g.:

```
withWebViewSingleton wv f = f (WebViewSingleton wv :: WebViewSingleton ())
```